common/k8stest/util_apps.go

```diff
@@ -419,6 +487,31 @@ func (dfa *FioApp) Cleanup() error {
 				return fmt.Errorf("verify replicas before deleting PVC failed: %v", err)
 			}
 		}
+	} else {
+		err := DeleteDeployment(dfa.DeployName, common.NSDefault)
+		if err != nil {
+			return fmt.Errorf("failed to delete fio deployment %s, err: %v", dfa.DeployName, err)
+		}
+		labels := "e2e-test=fio"
+		const sleepTime = 3
+		var runningStatus bool
+		for ix := 0; ix < (DefTimeoutSecs+sleepTime-1)/sleepTime; ix++ {
+			_, runningStatus, err = IsPodWithLabelsRunning(labels, common.NSDefault)
+			logf.Log.Info("IsPodWithLabelsRunning", "labels", labels, "error", err, "Running status", runningStatus)
+			if err != nil {
+				return fmt.Errorf("failed to get pod with label %s, err: %v", labels, err)
+			}
+			if !runningStatus {
+				break
+			}
+			time.Sleep(sleepTime * time.Second)
+		}
+		if runningStatus {	
+			err = dfa.DeletePod()
```

Comment:
Why do we need to delete pod if app is deployed as deployment ?

Reply:
When pod takes longer to get deleted this step would ensure it is getting deleted

Reply:
Don't delete deployment's pod explicitly , it should return error if pods are not getting deleted within given time period

Reply:
Fixed